### PR TITLE
Open gossip network to all nodes

### PR DESCRIPTION
**Pull request by rphmeier**
Previously, validator nodes only gossiped with each other. Now, all nodes will propagate consensus gossip messages. 

A CLI to disable gossip will be included in #460

### CODE REVIEW

**Comment by gavofyork** [comment]
needs a backport, too

File: substrate/network/src/consensus_gossip.rs [hunk 1]
```diff
@@ -116,6 +139,8 @@ impl<B: BlockT> ConsensusGossip<B> where B::Header: HeaderT<Number=u64> {
 
 	/// Handles incoming BFT message, passing to stream and repropagating.
 	pub fn on_bft_message(&mut self, protocol: &mut Context<B>, who: NodeIndex, message: message::LocalizedBftMessage<B>) {
+		if self.enabled { return }
```
Comment (gavofyork) [inline substrate/network/src/consensus_gossip.rs]: looks like this doesn't propagate if enabled - is that right?

**Comment by arkpar** [comment]
Actually, previously gossip was enabled for FULL nodes as well.
https://github.com/paritytech/polkadot/pull/359

**Comment by gavofyork** [comment]
so this doesn't do anything?

**Comment by rphmeier** [comment]
Ah, forgot about that one. I guess this is unnecessary.

**Pull request closed without merging**